### PR TITLE
feat: add agent service and message service

### LINKED ISSUE #10: Agent service
## Issue #10 — Agent service

Agent registration, heartbeat, and status management.

**Package:** `@caw/core`
**Depends on:** #2, #3, #4, #5

### Acceptance criteria

- [ ] `src/services/agent.service.ts`
- [ ] `register(name, runtime, role?, capabilities?, metadata?)` — creates agent, status `online`
- [ ] `heartbeat(agentId, currentTaskId?, status?)` — updates last_heartbeat
- [ ] `update(id, fields)` — update status, task, workspace, metadata
- [ ] `get(id)` — fetch agent details
- [ ] `list(filters?)` — filter by status, role, runtime
- [ ] `unregister(id)` — marks offline, releases any claimed tasks
- [ ] `getStale(timeoutMs)` — agents with heartbeat older than threshold
- [ ] Tests for registration, heartbeat staleness

### LINKED ISSUE #11: Message service
## Issue #11 — Message service

Inter-agent messaging with inbox pattern.

**Package:** `@caw/core`
**Depends on:** #10

### Acceptance criteria

- [ ] `src/services/message.service.ts`
- [ ] `send(senderId, recipientId, type, body, options?)` — creates message with thread support
- [ ] `broadcast(senderId, recipientFilter, type, body, options?)` — send to multiple agents
- [ ] `list(agentId, filters?)` — inbox with status/type/priority/thread filtering
- [ ] `get(id, markRead?)` — fetch single message
- [ ] `markRead(messageIds)` — batch mark as read
- [ ] `archive(messageIds)` — batch archive
- [ ] `countUnread(agentId, priorityFilter?)` — unread counts by priority
- [ ] Thread ID generation for conversations
- [ ] Tests for send/receive, threading, broadcast

### PR DESCRIPTION
**Pull request by travisgalloway**
Closes #10
Closes #11

## Summary
- Add `AgentService` with `register`, `heartbeat`, `update`, `get`, `list`, `unregister`, and `getStale` functions for managing agent lifecycle and stale agent detection
- Add `MessageService` with `send`, `broadcast`, `list`, `get`, `markRead`, `archive`, and `countUnread` functions for inter-agent inbox-style messaging with thread support
- 85 new tests (39 agent, 46 message) covering all functions, edge cases, and error paths
- Barrel exports added to `services/index.ts` and `core/index.ts` with aliased type re-exports to avoid name collisions
- No circular imports: `unregister()` writes directly to tasks table via SQL, mirroring the pattern used by `task.service.claim/release`

## Testing
- [x] Tests added/updated
- [x] All tests passing (249 total, 85 new)
- [x] Lint clean (Biome)
- [x] Full monorepo build passing

### CODE REVIEW

File: packages/core/src/services/agent.service.ts [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+export function list(db: DatabaseType, filters?: ListFilters): Agent[] {
+  const conditions: string[] = [];
+  const params: unknown[] = [];
+
+  if (filters?.status !== undefined) {
+    if (Array.isArray(filters.status)) {
+      const placeholders = filters.status.map(() => '?').join(', ');
+      conditions.push(`status IN (${placeholders})`);
+      params.push(...filters.status);
+    } else {
```
[the remark this answers is not in the document]
Comment (travisgalloway) [reply packages/core/src/services/agent.service.ts]: Fixed — added early return of `[]` when `filters.status` is an empty array. See 9b9a2d9.

File: packages/core/src/services/message.service.ts [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+
+  if (filters?.priority !== undefined) {
+    if (Array.isArray(filters.priority)) {
+      const placeholders = filters.priority.map(() => '?').join(', ');
+      conditions.push(`priority IN (${placeholders})`);
+      params.push(...filters.priority);
+    } else {
+      conditions.push('priority = ?');
+      params.push(filters.priority);
+    }
```
[the remark this answers is not in the document]
Comment (travisgalloway) [reply packages/core/src/services/message.service.ts]: Fixed — added empty array guards for all three array-accepting filters (`status`, `message_type`, `priority`). See 9b9a2d9.

File: packages/core/src/services/message.service.ts [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+
+  return { id, thread_id: threadId };
+}
+
+export function broadcast(db: DatabaseType, params: BroadcastParams): BroadcastResult {
+  const run = db.transaction(() => {
+    // Build agent query with filters
+    const conditions: string[] = ['id != ?'];
+    const queryParams: unknown[] = [params.sender_id];
+
```
[the remark this answers is not in the document]
Comment (travisgalloway) [reply packages/core/src/services/message.service.ts]: Fixed — added sender existence validation at the start of the `broadcast()` transaction, consistent with `send()`. See 9b9a2d9.
